contract/src/main.rs

```diff
@@ -1052,7 +1025,7 @@ pub extern "C" fn transfer() {
         NFTCoreError::InvalidReceiptName,
     );
 
-    let receipt_string = format!("{}-m-{}-p-{}", receipt_name, PAGE_SIZE, page_number);
+    let receipt_string = format!("{}-m-{}-p-{}", receipt_name, PAGE_SIZE, page_table_entry);
```

Comment:
i think you should pull this format into a utility and use it in all places you create the string to avoid accidental mismatch